packages/pd-manager/src/agent/PDManager.ts

```diff
@@ -0,0 +1,173 @@
+import { IAgentPlugin } from '@veramo/core'
+import {
+  DeleteDefinitionItemArgs,
+  GetDefinitionItemArgs,
+  GetDefinitionsItemArgs,
+  IPDManager,
+  PersistDefinitionArgs,
+  schema,
+  UpdateDefinitionItemArgs,
+} from '../index'
+import {
+  AbstractPDStore,
+  isPresentationDefinitionEqual,
+  NonPersistedPresentationDefinitionItem,
+  PresentationDefinitionItem,
+} from '@sphereon/ssi-sdk.data-store'
+import semver from 'semver/preload'
+
+// Exposing the methods here for any REST implementation
+export const pdManagerMethods: Array<string> = [
+  'pdmGetDefinition',
+  'pdmGetDefinitions',
+  'pdmAddDefinition',
+  'pdmUpdateDefinition',
+  'pdmRemoveDefinition',
+  'pdmPersistDefinition',
+]
+
+/**
+ * {@inheritDoc IPDManager}
+ */
+export class PDManager implements IAgentPlugin {
+  readonly schema = schema.IPDManager
+  readonly methods: IPDManager = {
+    pdmGetDefinition: this.pdmGetDefinition.bind(this),
+    pdmGetDefinitions: this.pdmGetDefinitions.bind(this),
+    pdmAddDefinition: this.pdmAddDefinition.bind(this),
+    pdmUpdateDefinition: this.pdmUpdateDefinition.bind(this),
+    pdmDeleteDefinition: this.pdmDeleteDefinition.bind(this),
+    pdmPersistDefinition: this.pdmPersistDefinition.bind(this),
+  }
+
+  private readonly store: AbstractPDStore
+
+  constructor(options: { store: AbstractPDStore }) {
+    this.store = options.store
+  }
+
+  /** {@inheritDoc IPDManager.pdmGetDefinition} */
+  private async pdmGetDefinition(args: GetDefinitionItemArgs): Promise<PresentationDefinitionItem> {
+    const { itemId } = args
+    return this.store.getDefinition({ itemId })
+  }
+
+  /** {@inheritDoc IPDManager.pdmGetDefinition} */
+  private async pdmGetDefinitions(args: GetDefinitionsItemArgs): Promise<Array<PresentationDefinitionItem>> {
+    const { filter } = args
+    return this.store.getDefinitions({ filter })
+  }
+
+  /** {@inheritDoc IPDManager.pdmAddDefinition} */
+  private async pdmAddDefinition(args: NonPersistedPresentationDefinitionItem): Promise<PresentationDefinitionItem> {
+    return this.store.addDefinition(args)
+  }
+
+  /** {@inheritDoc IPDManager.pdmUpdateDefinition} */
+  private async pdmUpdateDefinition(args: UpdateDefinitionItemArgs): Promise<PresentationDefinitionItem> {
+    return this.store.updateDefinition(args.definitionItem)
+  }
+
+  /** {@inheritDoc IPDManager.pdmDeleteDefinition} */
+  private async pdmDeleteDefinition(args: DeleteDefinitionItemArgs): Promise<boolean> {
+    return this.store.deleteDefinition(args).then((): boolean => true)
+  }
+
+  /** {@inheritDoc IPDManager.pdmPersistDefinition} */
+  private async pdmPersistDefinition(args: PersistDefinitionArgs): Promise<PresentationDefinitionItem> {
```

Comment:
this method is confusing, the other one persists as well? so what really is the point of this method?

Reply:
Details are written in the help text:
```
  /**
   * Persist a presentation definition.
   * It has version control logic which will add or update presentation definition records and has settings for automatic version numbering.
   * @param args
   */
```

This is also how the function was called in PresentationExchange If you know a better name, please let me know.

Reply:
This naming choice is very confusing. 
If you are creating any type of object, it is being persisted somewhere.
Theferore, I don't expect to see both an `add` and a `persist` functions that return the same object (PresentationDefinitionItem).
Also, why do you need to create a PD object without version control logic?